viewer/core/src/public/migration/restrictViewerOptionsToDevice.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import { Cognite3DViewerOptions } from './types';
+
+import log from '@reveal/logger';
+import { assertNever, isMobileOrTablet } from '@reveal/utilities';
+
+/**
+ * Modifies options to disable features we don't support on the current device. The following options might be modified:
+ * - SSAO - disabled for mobile/tablet
+ * - Anti-aliasing - for mobile/tablet we will fall back to FXAA when MSAA is enabled.
+ * @param options
+ */
+export function restrictViewerOptionsToDevice(options: Cognite3DViewerOptions): void {
+  options.antiAliasingHint = restrictAntialiasOption(options.antiAliasingHint);
+  options.ssaoQualityHint = restrictSsaoOption(options.ssaoQualityHint);
+}
+
+type AntiAliasingHint = Cognite3DViewerOptions['antiAliasingHint'] | undefined;
+type SsaoQualityHint = Cognite3DViewerOptions['ssaoQualityHint'] | undefined;
+
+function restrictAntialiasOption(antiAliasingHint: AntiAliasingHint): AntiAliasingHint {
+  if (!isMobileOrTablet()) {
+    return antiAliasingHint;
+  }
+
+  switch (antiAliasingHint) {
+    case 'msaa2':
+    case 'msaa4':
+    case 'msaa8':
+    case 'msaa16':
+    case 'msaa2+fxaa':
+    case 'msaa4+fxaa':
+    case 'msaa8+fxaa':
+    case 'msaa16+fxaa':
+      log.warn(`Anti-aliasing mode '${antiAliasingHint}' is not supported on mobile devices, falling back to 'fxaa'`);
+      return 'fxaa';
+
+    case 'disabled':
+    case 'fxaa':
+    case undefined:
+      return antiAliasingHint;
+
+    default:
+      assertNever(antiAliasingHint);
+  }
+}
+
+function restrictSsaoOption(ssaoQualityHint: SsaoQualityHint): SsaoQualityHint {
+  if (!isMobileOrTablet()) {
+    return ssaoQualityHint;
+  }
+
+  switch (ssaoQualityHint) {
+    case 'medium':
+    case 'high':
+    case 'veryhigh':
+      log.warn(`SSAO is not supported on mobile devices, disabling`);
+      return 'disabled';
+
+    case 'disabled':
+    case undefined:
+      return ssaoQualityHint;
+
+    default:
+      assertNever(ssaoQualityHint);
+  }
+}
```

Comment:
We already have a `createRevealManagerOptions`, that has `determineAntiaAlisaingMode` and `determineSsaoParamateres`. It makes more sense to me that it should be handled here instead of as a pre-create options step.

Reply:
We might - but I wanted to separate the concern and let `createRevealManagerOptions` simply do as it's told. Again, I'll change this if you think this is just extra noise and unnecessary separation.

Reply:
In my opinion this would not brake SOC, it would also simplify the code (or at least share some code), as we are technically setting the options twice with the current approach. The way I see it is that `createRevealManagerOptions`s job is to do this filtering / defaulting of the options. It also becomes clearer when reading code (one point of altering the input options data vs. two).

Reply:
Not sure I think it's better now, but changed.